### PR TITLE
Async Stacktrace improvements 

### PR DESCRIPTION
**Pull request by pengweiqhca**
#101

### CODE REVIEW

**Comment by bruno-garcia** [comment]
@pengweiqhca running the samples I don't see any difference on the resulting stack trace.
As I mentioned on #101 it seems it only affects the `ToString` of the exception.

Could you please show some examples where the result at Sentry is in fact improved?

Also, I wouldn't want to create a dependency to `Ben.Demystifier` from `Sentry` but rather perhaps create a new package like `Sentry.Ben.Demystifier` or some generic name like `Sentry.Enhanced` where we can add other dependencies.

**Comment by pengweiqhca** [comment]
@bruno-garcia Need async method

File: src/Sentry.Ben.Demystifier/Sentry.Ben.Demystifier.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <Description>Official Microsoft.Extensions.Logging integration for Sentry - Open-source error tracking that helps developers monitor and fix crashes in real time.</Description>
```
[the remark this answers is not in the document]
Comment (pengweiqhca) [reply src/Sentry.Ben.Demystifier/Sentry.Ben.Demystifier.csproj]: Modified. See the next commit.

**Comment by pengweiqhca** [comment]
@bruno-garcia  This is the raw data
tim 20181015063325

**Comment by bruno-garcia** [comment]
@pengweiqhca sorry for the delay on merging this. I haven't had much time to look into this unfortunately.
One this I'm still unsure is whether to publish this as `Sentry.Ben.Demystifier` or create some package named `Sentry.Extended` or hopefully a better name than that. That would allow us to add more dependencies later to get other features and enhancements. 
Basically splitting in 2 packages, the main package with main functionality, and a second one which has more dependencies but gives you more functionality.

I've been holding off on this decision because when getting involved with Xamarin and Unity3d, part of the current `Sentry` package might be pulled out into a `Sentry.Core` package, that holds the bits that are used in any SDK we build and the current `Sentry` SDK would expose what it does now but also we could add more dependencies to it directly like the one you've added. But in that case we need to offer a lightweight version for people who cannot or do not want to have many dependencies.

**Comment by lunim** [comment]
Any updates?

**Comment by bruno-garcia** [comment]
@lunim @pengweiqhca At this point, changes in stacktrace would change how events get grouped in Sentry. If we roll out this package under Sentry, anyone replacing `Sentry` with `Sentry.Ben.Demystifier` would be spammed with notifications of new events.
We're working on new grouping which could help in this case but in any case, I think this would be a great community addition. So not at this point published under Sentry where we need to support and document under _docs.sentry.io_ but simply community maintained. Maybe @pengweiqhca is interested in publishing under his account? Or a github org for Sentry .NET Community packages or something like that.

Worse case, until we sort this out, you could still just build this branch locally and add the package to your application.

**Comment by pengweiqhca** [comment]
@bruno-garcia I create new PR https://github.com/getsentry/sentry-dotnet/pull/178

**Pull request closed without merging**